File: src/main/scala/arriba/scala/fix/disruptor/SerializedFieldsToRingBufferEntryAdapter.scala
package arriba.scala.fix.disruptor

import arriba.scala.common.MessageToRingBufferEntryAdapter
import arriba.scala.fix.SerializedField

final class SerializedFieldsToRingBufferEntryAdapter extends MessageToRingBufferEntryAdapter[List[SerializedField], FixMessageEntry] {
  override def adapt(serializedFields: List[SerializedField], entry: FixMessageEntry) {
    entry.setSerializedFields(serializedFields)
  }
}

File: src/main/java/arriba/fix/disruptor/SerializedFieldsToRingBufferEntryAdapter.java
package arriba.fix.disruptor;

import java.util.List;

import arriba.scala.common.MessageToRingBufferEntryAdapter;
import arriba.fix.SerializedField;

public final class SerializedFieldsToRingBufferEntryAdapter implements MessageToRingBufferEntryAdapter<List<SerializedField>, FixMessageEntry>{

    public void adapt(final List<SerializedField> serializedFields, final FixMessageEntry entry) {
        entry.setSerializedFields(serializedFields);
    }
}
